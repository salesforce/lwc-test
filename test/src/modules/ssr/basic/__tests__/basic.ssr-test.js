/*
 * Copyright (c) 2018, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */
import { renderComponent } from 'lwc';
import Basic from 'ssr/basic';

it('renders a basic component and saves formatted snapshot', () => {
    const renderedComponent = renderComponent('x-basic', Basic, { msg: 'Hello world' });

    expect(renderedComponent).toMatchSnapshot();
});

it('renders a basic component and saves inline formatted snapshot', () => {
    const renderedComponent = renderComponent('x-basic', Basic, { msg: 'Hello world' });

    expect(renderedComponent).toMatchInlineSnapshot(`
        <x-basic>
          <template shadowrootmode="open">
            <style type="text/css">
              h1 {color: red;}
            </style>
            <h1>
              Basic, Hello world
            </h1>
          </template>
        </x-basic>
    `);
});

it('runs on node environment', () => {
    expect(() => document).toThrow();
});
