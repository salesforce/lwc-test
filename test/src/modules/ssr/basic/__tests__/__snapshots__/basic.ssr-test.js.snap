// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic component and saves formatted snapshot 1`] = `
<x-basic>
  <template shadowrootmode="open">
    <style type="text/css">
      h1 {color: red;}
    </style>
    <h1>
      Basic, Hello world
    </h1>
  </template>
</x-basic>
`;
